### PR TITLE
PIM-4024: Fix for metric and price denormalizer

### PR DESCRIPTION
**Pull request by willy-ahva**
| Q                    | A
| -------------------- | ---
| Bug fix?             | y
| New feature?         | n
| BC breaks?           | y
| CI currently passes? | y
| Tests pass?          | y
| Scenarios pass?      | y
| Checkstyle issues?*  | n
| PMD issues?**        | n
| Changelog updated?   | y
| Fixed tickets        | PIM-4024
| DB schema updated?   | n
| Migration script?    | n
| Doc PR               |

### CODE REVIEW

File: src/Pim/Bundle/TransformBundle/Denormalizer/Flat/ProductValue/MetricDenormalizer.php [hunk 1]
```diff
@@ -36,9 +36,7 @@ public function __construct(array $supportedTypes, MetricFactory $factory)
      */
     public function denormalize($data, $class, $format = null, array $context = array())
     {
-        if ($data === null || $data === '') {
-            return null;
-        }
+        $data = '' === $data ? null : $data;
```
Comment (solivier) [inline src/Pim/Bundle/TransformBundle/Denormalizer/Flat/ProductValue/MetricDenormalizer.php]: is it normal that before you had an early return and now you're just returning null and you're executing the rest of the method ?

File: src/Pim/Bundle/TransformBundle/Denormalizer/Flat/ProductValue/PricesDenormalizer.php [hunk 2]
```diff
@@ -36,9 +36,7 @@ public function __construct(array $supportedTypes, ProductBuilder $productBuilde
      */
     public function denormalize($data, $class, $format = null, array $context = array())
     {
-        if ($data === null || $data === '') {
-            return null;
-        }
+        $data = '' === $data ? null : $data;
```
Comment (solivier) [inline src/Pim/Bundle/TransformBundle/Denormalizer/Flat/ProductValue/PricesDenormalizer.php]: same here

File: src/Pim/Bundle/TransformBundle/Denormalizer/Flat/ProductValue/MetricDenormalizer.php (hunk 1 above)
Comment (willy-ahva) [inline src/Pim/Bundle/TransformBundle/Denormalizer/Flat/ProductValue/MetricDenormalizer.php]: That's the purpose of this fix. Stop to return null ;)

Comment (skeleton) [inline src/Pim/Bundle/TransformBundle/Denormalizer/Flat/ProductValue/MetricDenormalizer.php]: yoda code style please

File: src/Pim/Bundle/TransformBundle/Denormalizer/Flat/ProductValue/PricesDenormalizer.php (hunk 2 above)
Comment (skeleton) [inline src/Pim/Bundle/TransformBundle/Denormalizer/Flat/ProductValue/PricesDenormalizer.php]: idem, yoda code style please

**Pull request merged**